atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/ValueType.java

```diff
@@ -464,9 +463,7 @@ public String convertToJava(byte[] value, int offset) {
 
         @Override
         public byte[] convertFromJson(String jsonValue) {
-            Object s = JSONValue.parse(jsonValue);
-            Preconditions.checkArgument(s instanceof String, "%s must be a json string", jsonValue);
-            return convertFromString((String) s);
+            return convertFromString(ValueType.readJson(jsonValue, String.class));
```

Comment:
To make sure I understand: before, you would get `SafeIllegalArgumentException`s if this was not a string, but now you just get a plain `RuntimeException` that wraps an `IOException`. Is my understanding correct, and is this behaviour OK?

Reply:
That's mostly correct (strictly speaking: it used Google, not Palantir Preconditions, so you'd just get a regular `IllegalArgumentException`). That's a good call: I _think_ that is fine, but I'll change it back.